views/default/input/date.php

```diff
@@ -48,8 +52,20 @@
 	$vars['value'] = gmdate('Y-m-d', $vars['value']);
 }
 
+$datepicker_options = elgg_extract('datepicker_options', $vars);
```

Comment:
Just make the default an empty string here. Then you don't need the ternary: just json_encode either way.